x-pack/plugins/ingest_manager/server/services/epm/archive/index.ts

```diff
@@ -0,0 +1,320 @@
+
+  datasetPaths.forEach((datasetPath) => {
+    const manifestFile = `${pkgName}-${pkgVersion}/dataset/${datasetPath}/manifest.yml`;
+    const manifestBuffer = cacheGet(manifestFile);
+    if (!paths.includes(manifestFile) || !manifestBuffer) {
+      throw new PackageInvalidArchiveError(
+        `No manifest.yml file found for dataset '${datasetPath}'`
+      );
+    }
+
+    let manifest;
+    try {
+      manifest = yaml.load(manifestBuffer.toString());
+    } catch (error) {
+      throw new PackageInvalidArchiveError(
+        `Could not parse package manifest for dataset '${datasetPath}': ${error}.`
+      );
+    }
+
+    const { title: datasetTitle, release, ingest_pipeline: ingestPipeline, type } = manifest;
+    if (!(datasetTitle && release && type)) {
+      throw new PackageInvalidArchiveError(
+        `Invalid manifest for dataset '${datasetPath}': one or more fields missing of 'title', 'release', 'type'`
+      );
+    }
+    const streams = parseAndVerifyStreams(manifest, datasetPath);
+
+    // default ingest pipeline name see https://github.com/elastic/package-registry/blob/master/util/dataset.go#L26
+    return datasets.push({
+      name: `${pkgName}.${datasetPath}`,
+      title: datasetTitle,
+      release,
+      package: pkgName,
+      ingest_pipeline: ingestPipeline || 'default',
+      path: datasetPath,
+      type,
+      streams,
+    });
+  });
+
+  return datasets;
+}
+
+function parseAndVerifyStreams(manifest: any, datasetPath: string): RegistryStream[] {
+  const streams: RegistryStream[] = [];
+  const manifestStreams = manifest.streams;
+  if (manifestStreams && manifestStreams.length > 0) {
+    manifestStreams.forEach((manifestStream: any) => {
+      const {
+        input,
+        title: streamTitle,
+        description,
+        enabled,
+        vars: manifestVars,
+        template_path: templatePath,
+      } = manifestStream;
+      if (!(input && streamTitle)) {
+        throw new PackageInvalidArchiveError(
+          `Invalid manifest for dataset ${datasetPath}: stream is missing one or more fields of: input, title`
+        );
+      }
+      const vars = parseAndVerifyVars(manifestVars, `dataset ${datasetPath}`);
+      // default template path name see https://github.com/elastic/package-registry/blob/master/util/dataset.go#L143
+      streams.push({
+        input,
+        title: streamTitle,
+        description,
+        enabled,
+        vars,
+        template_path: templatePath || 'stream.yml.hbs',
+      });
+    });
+  }
+  return streams;
+}
+
+function parseAndVerifyVars(manifestVars: any[], location: string): RegistryVarsEntry[] {
+  const vars: RegistryVarsEntry[] = [];
+  if (manifestVars && manifestVars.length > 0) {
+    manifestVars.forEach((manifestVar) => {
+      const {
+        name,
+        title: varTitle,
+        description,
+        type,
+        required,
+        show_user: showUser,
+        multi,
+        def,
+        os,
```

Comment:
If I remember correctly, we don't support os at the moment.

Reply:
We have a few leftovers in our types (`internal` on the main package manifest is another one). I'd rather remove them in a separate PR, just to reduce the risk of additional breakage.